packages/main/src/DateRangePicker.js

```diff
@@ -0,0 +1,473 @@
+		const calendar = this.responsivePopover.querySelector(`#${this._id}-calendar`);
+		const dayPicker = calendar.shadowRoot.querySelector(`#${calendar._id}-daypicker`);
+		dayPicker.addEventListener("item-mouseover", this._itemMouseoverHandler);
+		// dayPicker.addEventListener("item-keydown", this._itemKeydownHandler);
+		dayPicker.addEventListener("daypickerrendered", this._borderReachedWithHovering);
+
+		this._cleanHoveredAttributeFromVisibleItems(dayPicker);
+	}
+
+	_itemMouseoverHandler(event) {
+		const dayItems = event.target.shadowRoot.querySelectorAll(".ui5-dp-item");
+		const firstDateTimestamp = this._selectedDates[0];
+		const lastDateTimestamp = event.detail.target.parentElement.dataset.sapTimestamp;
+
+		for (let i = 0; i < dayItems.length; i++) {
+			if ((dayItems[i].dataset.sapTimestamp < firstDateTimestamp && dayItems[i].dataset.sapTimestamp > lastDateTimestamp)
+				|| (dayItems[i].dataset.sapTimestamp > firstDateTimestamp && dayItems[i].dataset.sapTimestamp < lastDateTimestamp)) {
+				dayItems[i].setAttribute("hovered", "");
+			} else {
+				dayItems[i].removeAttribute("hovered");
+			}
+		}
+	}
+
+	_itemKeydownHandler(event) {
+		const dayItems = event.target.shadowRoot.querySelectorAll(".ui5-dp-item"),
+			firstDateTimestamp = this._selectedDates[0],
+			oneDayTimestamp = 24 * 60 * 60,
+			sevenDaysTimestamp = 7 * 24 * 60 * 60;
+		let lastDateTimestamp = Number(event.detail.target.dataset.sapTimestamp);
+
+		if (isUp(event.detail)) {
+			lastDateTimestamp -= sevenDaysTimestamp;
+		} else if (isDown(event.detail)) {
+			lastDateTimestamp += sevenDaysTimestamp;
+		} else if (isLeft(event.detail)) {
+			lastDateTimestamp -= oneDayTimestamp;
+		} else if (isRight(event.detail)) {
+			lastDateTimestamp += oneDayTimestamp;
+		}
+
+		for (let i = 0; i < dayItems.length; i++) {
+			if ((dayItems[i].dataset.sapTimestamp < firstDateTimestamp && dayItems[i].dataset.sapTimestamp > lastDateTimestamp)
+				|| (dayItems[i].dataset.sapTimestamp > firstDateTimestamp && dayItems[i].dataset.sapTimestamp < lastDateTimestamp)) {
+				dayItems[i].setAttribute("hovered", "");
+			} else {
+				dayItems[i].removeAttribute("hovered");
+			}
+		}
+	}
+
+	_borderReachedWithHovering(event) {
+		if (!event.detail.focusedItemIndex) {
+			return;
+		}
+
+		const dayItems = event.target.shadowRoot.querySelectorAll(".ui5-dp-item");
+		const firstDateTimestamp = this._selectedDates[0];
+		const lastDateTimestamp = dayItems[event.detail.focusedItemIndex].dataset.sapTimestamp;
+
+		for (let i = 0; i < dayItems.length; i++) {
+			if ((dayItems[i].dataset.sapTimestamp < firstDateTimestamp && dayItems[i].dataset.sapTimestamp > lastDateTimestamp)
+				|| (dayItems[i].dataset.sapTimestamp > firstDateTimestamp && dayItems[i].dataset.sapTimestamp < lastDateTimestamp)) {
+				dayItems[i].setAttribute("hovered", "");
+			} else {
+				dayItems[i].removeAttribute("hovered");
+			}
+		}
+	}
+
+	_splitValueByDelimiter(value, delimiter) {
```

Comment:
I would rather skip the second parameter, because you are always calling the method with this.delimiter as delimiter. This will simplify the implementation of the method as well.